dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/apps/newTimer/EjbSessionBean.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.ejb.apps.newtimer;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.EJB;
+import javax.ejb.Local;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.ejb.Timeout;
+import javax.ejb.Timer;
+import javax.ejb.TimerConfig;
+import javax.ejb.TimerService;
+import javax.inject.Inject;
+
+import javax.enterprise.context.ContextNotActiveException;
+
+/**
+ * Session Bean implementation class EjbSessionBean
+ */
+@Stateless
+@Local(EjbSessionBeanLocal.class)
+@LocalBean
+@MyCDIInterceptorBinding
+public class EjbSessionBean implements EjbSessionBeanLocal {
+    private static final long TIMEOUT = 1;
+
+    @Resource
+    TimerService timerService;
+
+    @Inject RequestScopedBean rsb;
+
+    @Inject SessionScopedBean ssb;
+
+    private int requestScopedIndex = -1;
+    public static Boolean canAccessRequestScope = null;
+    public static Boolean seperateRequestScopes = null;
+    public static Boolean sessionScopeInactive = null;
+
+
+    @Timeout
+    void ping(Timer timer) {
+        //Test One. Is a request scope active in the context of an EJB timer?
+        try {
+            if (rsb.toString() != null) {
+                canAccessRequestScope = true;
+            } else {
+                canAccessRequestScope = false;
+            }
+        } catch (ContextNotActiveException e) {
+            canAccessRequestScope = false;
+            throw e;
+        }
```

Comment:
Throwing the exception here prevents the rest of the test from running, though I guess that doesn't matter too much because each of the boolean fields are asserted in turn within one test.

As this is a `@Timeout` method, I guess the `ContextNotActiveException` will get logged/FFDC'd, so I guess that could be useful?

Was that what you were intending?